app/main.py

```diff
@@ -1,25 +1,76 @@
+from __future__ import annotations
+
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: any, name: str) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: Visitor, owner: any) -> str:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: Visitor, value: int) -> None:
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError
```

Comment:
it is better to write some message with raising error

Reply:
Ok, but I always handle this raise in the function can_access

Reply:
But it also should work by itself